Test/Unit/Model/Api/ShippingMethodsTest.php

```diff
@@ -0,0 +1,504 @@
+        ];
+        $quote = $this->getMockBuilder(Quote::class)
+            ->setMethods($quoteMethods)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $quote->method('getId')
+            ->willReturn($quoteId);
+        $quote->method('getBoltParentQuoteId')
+            ->willReturn($parentQuoteId);
+        $quote->method('getSubtotal')
+            ->willReturn(100);
+        $quote->method('getAllVisibleItems')
+            ->willReturn([$quoteItem]);
+        $quote->method('getAppliedRuleIds')
+            ->willReturn('2,3');
+        $quote->method('isVirtual')
+            ->willReturn(false);
+        $quote->method('getShippingAddress')
+            ->willReturn($shippingAddress);
+
+        $cartHelper = $this->getMockBuilder(CartHelper::class)
+            ->setMethods(['getQuoteById'])
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $cartHelper->method('getQuoteById')
+            ->with($quoteId)
+            ->willReturn($quote);
+
+        $configHelper = $this->getMockBuilder(ConfigHelper::class)
+            ->setMethods(['getPrefetchShipping', 'getPrefetchAddressFields'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $configHelper->method('getPrefetchShipping')
+            ->willReturn(true);
+        $configHelper->method('getPrefetchAddressFields')
+            ->willReturn('');
+
+        $methods = ['sendErrorResponse', 'proceedWithHook', 'checkCartItems', 'getQuoteById',
+            'notifyException', 'validateQuote', 'loadSessionByQuote', 'throwQuoteIdException',
+            'validateAddressData', 'shippingEstimation'
+        ];
+        $this->currentMock = $this->getMockBuilder(BoltShippingMethods::class)
+            ->setMethods($methods)
+            ->setConstructorArgs([
+                $this->hookHelper,
+                $this->regionModel,
+                $this->factoryShippingOptionsMock,
+                $this->shippingTaxInterfaceFactory,
+                $cartHelper,
+                $this->totalsCollector,
+                $this->converter,
+                $this->shippingOptionInterfaceFactory,
+                $this->bugsnag,
+                $this->logHelper,
+                $this->errorResponse,
+                $this->response,
+                $configHelper,
+                $this->request,
+                $this->cache,
+                $this->priceHelper,
+                $this->sessionHelper
+            ])
+            ->getMock();
+
+        $this->currentMock->method('validateAddressData')
+            ->willReturnSelf();
+
+        $option = new \Bolt\Boltpay\Model\Api\Data\ShippingOption();
+        $option
+            ->setService('Flat Rate - Fixed')
+            ->setCost(5600)
+            ->setReference('flatrate_flatrate')
+            ->setTaxAmount(0)
+        ;
+
+        $shippingOptionData = [$option];
+
+        $this->currentMock->method('shippingEstimation')
```

Comment:
can you mock one-level deeper? Idea of adding unit test is to make sure our logic is working for apple pay. If someone revert https://github.com/BoltApp/bolt-magento2/pull/185 unit test should fail. However current test would pass because we mock shippingEstimation